text/0000-light-dom-scoped-styles.md

```diff
@@ -0,0 +1,233 @@
+---
+title: Light DOM Scoped Styles
+status: DRAFTED
+created_at: 2021-05-14
+updated_at: YYYY-MM-DD
+champion: Nolan Lawson (nolanlawson), Abdulsattar Mohammed (abdulsattar)
+pr: https://github.com/salesforce/lwc-rfcs/pull/50
+---
+
+# Light DOM Scoped Styles
+
+## Summary
+
+This proposal adds a mechanism for light DOM components ([#44](https://github.com/salesforce/lwc-rfcs/pull/44)) to have co-located CSS files that only apply to that component.
+
+## Basic example
+
+Since there is already `*.css` for unscoped light DOM CSS files, we add `*.scoped.css` for scoped light DOM CSS:
+
+```html
+<!-- x/foo.html -->
+<template>
+  <div>Hello</div>
+</template>
+```
+
+```js
+// x/foo.js
+import { LightningElement } from 'lwc'
+
+export default class Foo extends LightningElement {
+  static shadow = false
+}
+```
+
+```css
+/* x/foo.scoped.css */
+div {
+  color: green;
+}
+```
+
+The result will be:
+
+```html
+<x-app>
+  <style class="x-foo_foo" type="text/css">
+    div.x-foo_foo { color: green; }
+  </style>
+  <div class="x-foo_foo">Hello</div>
+</x-app>
+```
+
+## Motivation
+
+For Light DOM ([#44](https://github.com/salesforce/lwc-rfcs/pull/44)), the default assumption is that styles are unscoped. In other words, a light DOM component essentially just contains `<style>` tags that are inserted into the DOM _in situ_. (The actual implementation may be slightly different, but this is the basic mental model.)
+
+Many frameworks, however, have a concept of _scoped styles_, even without using shadow DOM. These offer good developer ergonomics, because developers can concentrate on the CSS co-located with a particular component, without worrying how that CSS might affect other components.
+
+Prior art:
+
+- [Vue: `<style scoped>`](https://vue-loader.vuejs.org/guide/scoped-css.html)
+- [Svelte: scoped styles](https://svelte.dev/docs#style)
+- [React: CSS Modules in `create-react-app`](https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/)
+- [Stencil: scoped styles in light DOM](https://stenciljs.com/docs/styling#scoped-css)
+- [BEM: Block, Element, Modifier](https://en.bem.info/)
+
+## Detailed design

```

Comment:
> I would personally favor a strong signal

This seems unnecessary to me, since we don't already do this for existing `*.css` files. Developers can include or omit the `*.css` file, and the compiler will pick it up automatically. Why shouldn't it be the same for `*.scoped.css` files? If we're going to have a new system for scoped CSS, I would prefer extending it back to regular CSS as well – even if it's a breaking change, it would be better for consistency.

I admit `<style scoped>` would be more preferable from the perspective of maintaining Rollup/Jest/etc. plugins, but again, it's a totally different system from existing CSS, and it wouldn't be clear to developers why they need one system for one and another system for the other. Also it would raise the question of why we don't allow "single file" components ala Vue/Svelte - e.g. a `<script>` tag containing the JS directly inlined in the HTML.